soccer/src/soccer/strategy/agent/agent_action_client.cpp

```diff
@@ -62,11 +62,26 @@ void AgentActionClient::get_task() {
         }
     }
 
-    auto task = current_position_->get_task();
-    if (task != last_task_) {
-        /* SPDLOG_INFO("sending new task"); */
-        last_task_ = task;
-        send_new_goal();
+    auto optional_task = current_position_->get_task();
+    if (optional_task.has_value()) {
+        rj_msgs::msg::RobotIntent task = optional_task.value();
+
+        // note that this comparison uses the ROS built-in msg type
+        // so any custom operator== overloads written don't apply
+        // TODO(Kevin): fix this by making Positions return RobotIntent structs, not msgs
+        if (task != last_task_) {
+            /* if (robot_id_ == 0) { */
+            /*     SPDLOG_INFO("sending new task {}", task.motion_command.name); */
+            /* } */
+            last_task_ = task;
+            send_new_goal();
+        } else {
+            /* if (robot_id_ == 0) { */
```

Comment:
we can delete these commented out lines when merging.

Reply:
ah, this is superseded by the same lines in #1988 (see [here](https://github.com/RoboJackets/robocup-software/pull/1988/files#diff-7a4b491bf7b4c369564c340fdcc53ef9ceb30d1668a41e4063ac34ee174704b4R72-R75)) so it won't last long on ros2 anyways 😆